components/nexus-orient/src/main/java/org/sonatype/nexus/orient/PersistentDatabaseInstanceRule.java

```diff
@@ -0,0 +1,168 @@
+import org.junit.rules.ExternalResource;
+import org.junit.runners.model.MultipleFailureException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.bridge.SLF4JBridgeHandler;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * JUnit rule to provide a persistent {@link DatabaseInstance} and related components.
+ *
+ * @since 3.0
+ */
+public class PersistentDatabaseInstanceRule
+  extends ExternalResource
+{
+  static {
+    try {
+      SLF4JBridgeHandler.removeHandlersForRootLogger();
+      SLF4JBridgeHandler.install();
+    } catch (LinkageError e) {
+      // no-op, jul-to-slf4j not installed
+    }
+  }
+
+  private static final Logger log = LoggerFactory.getLogger(PersistentDatabaseInstanceRule.class);
+
+  private final String name;
+
+  private MinimalDatabaseServer server;
+
+  private PersistentDatabaseManager manager;
+
+  private DatabaseInstance instance;
+
+  public PersistentDatabaseInstanceRule(final String name) {
+    this.name = checkNotNull(name);
+  }
+
+  public MinimalDatabaseServer getServer() {
+    checkState(server != null);
+    return server;
+  }
+
+  public DatabaseManager getManager() {
+    checkState(manager != null);
+    return manager;
+  }
+
+  public DatabaseInstance getInstance() {
+    checkState(instance != null);
+    return instance;
+  }
+
+  @Override
+  protected void before() throws Throwable {
+    log.info("Preparing database instance: {}", name);
+
+    server = new MinimalDatabaseServer();
+    server.start();
+
+    manager = new PersistentDatabaseManager();
+    manager.start();
+
+    instance = manager.instance(name);
+
+    log.info("Database instance prepared");
+  }
+
+  @Override
+  protected void after() {
+    log.info("Cleaning up database instance: {}", name);
+
+    instance = null;
+
+    // capture any errors so we can propagate to test harness
+    List<Throwable> errors = Lists.newArrayListWithCapacity(2);
+
+    if (manager != null) {
+      try {
+        manager.stop();
+      }
+      catch (Exception e) {
+        log.error("Failed to stop database manager", e);
+        errors.add(e);
+      }
+      manager = null;
+    }
+
+    if (server != null) {
+      try {
+        server.stop();
+      }
+      catch (Exception e) {
+        log.error("Failed to stop database server", e);
+        errors.add(e);
+      }
+      server = null;
+    }
+
+    if (!errors.isEmpty()) {
+      log.error("Failed to clean up database instance");
+      throw Throwables.propagate(new MultipleFailureException(errors));
+    }
+
+    log.info("Database instance cleaned up");
+  }
+
+  class PersistentDatabaseManager
+      extends DatabaseManagerSupport {
+
+    private final File databasesDirectory;
+
+    PersistentDatabaseManager() {
+      File targetDir = new File(this.getClass().getProtectionDomain().getCodeSource().getLocation().getFile()).getParentFile();
```

Comment:
probably can avoid all this and use TemporaryFolder, unless there's a compelling reason to keep the file system around after the test exits